bullet_train-obfuscates_id/app/validators/restricted_paths_validator.rb

```diff
@@ -0,0 +1,7 @@
+class RestrictedPathsValidator < ActiveModel::Validator
+  def validate(record)
+    if record.class.restricted_paths.include?(record.slug)
+      record.errors.add record.class.slug_attribute, :restricted_path
```

Comment:
Here we may also need to validate that the slug doesn't match any of the controller actions for this model. For instance if someone tried to use the slug `new` for a team, then that would cause a collision for the path `/teams/new`. We'd always want that to point to `new_team_path` and not to `team_path(slug: 'new')`.

[This article](https://nts.strzibny.name/restricted-paths-validator-url-slugs/) suggests that something like `TeamsController.action_methods` would give us a good list to validate against.

For our `Account::TeamsController` the `action_methods` method returns this big ol' list:

```ruby
Account::TeamsController.action_methods
=>
#<Set:
 {"team_params",
  "switch_to",
  "show",
  "index",
  "create",
  "edit",
  "destroy",
  "update",
  "new",
  "ensure_onboarding_is_complete",
  "assign_select_options",
  "create_model_if_new",
  "create_models_if_new",
  "ensure_backing_models_on",
  "assign_checkboxes",
  "assign_date",
  "assign_date_and_time",
  "assign_boolean",
  "load_team",
  "managing_account?",
  "ensure_onboarding_is_complete_and_set_next_step",
  "set_last_seen_at",
  "adding_team?",
  "accepting_invitation?",
  "adding_user_email?",
  "adding_user_details?",
  "switching_teams?",
  "set_current_user",
  "invited?",
  "show_sign_up_options?",
  "after_sign_up_path_for",
  "enforce_invitation_only",
  "only_allow_path",
  "delegate_json_to_api",
  "current_team",
  "current_membership",
  "current_locale",
  "set_locale",
  "layout_by_resource",
  "set_sentry_context"}
>
```

That seems like too much. [This StackOverflow answer](https://stackoverflow.com/a/53528975/631187) suggests a method like this:

```ruby
def actions_for_controller(controller_path)
  route_defaults = Rails.application.routes.routes.map(&:defaults)
  route_defaults = route_defaults.select { |x| x[:controller] == controller_path }
  route_defaults.map { |x| x[:action] }.uniq
end
```

Which produces a much more reasonable list:

```ruby
actions_for_controller('account/teams')
=> [
  "index",
  "create",
  "new",
  "edit",
  "show",
  "update",
  "destroy",
  "switch_to"
]
```

I'm not sure that we'd _need_ to prevent `member` actions like `edit` and `show` from being used as a slug, though allowing them might result in weird URLs like `/teams/show/edit` (the `edit` action for the team with a slug of `show`).

Determining which controller to inspect for a given model may be difficult.